sklearn/linear_model/_omp.py

```diff
@@ -646,11 +650,32 @@ def fit(self, X, y):
         self : object
             returns an instance of self.
         """
+        if self.normalize == 'deprecate':
+            _normalize = True
+        else:
+            _normalize = self.normalize
+
+        if not _normalize:
```

Comment:
I'm not sure we can go from `normalize=True` to removing the parameter with no normalization in one single cycle: with the current implementation, there is no way to avoid the warnings.

It seems that we should first change the default from True to False, and then remove the parameter?